modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -79,11 +81,14 @@ class S3BlobContainer extends AbstractBlobContainer {
 
     private final S3BlobStore blobStore;
     private final String keyPath;
+    @Nullable
+    private final Purpose purpose;
```

Comment:
I would much prefer this not to be nullable, let's represent the cases that would be null as a proper enum value instead. Indeed at some point in the future I think we'll want to [distinguish metadata and data operations](https://github.com/elastic/elasticsearch/issues/81352) too, and this seems like a good mechanism for that.

Reply:
Thanks David. I replaced `null` with `Purpose.Generic` in 0a399421c41e2cc96f0410e6a58414fb0707a249